assets/questions/palindrome.js

```diff
@@ -0,0 +1,166 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question data for Balanced Palindrome.
+ */
+
+globalData.questions['palindrome'] = {  // eslint-disable-line dot-notation
+  title: 'Balanced Palindrome',
+  starterCode: {
+    python:
+`def isPalindrome(s):
+    return False
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+
+    @classmethod
+    def createPalindrome(cls, character_set, input_size):
+        import random
+
+        space_number = random.randint(0, input_size)
+        character_number = input_size - space_number
+
+        palindrome = []
+        while character_number > 0:
+            character = random.choice(character_set)
+            if character_number == 1:
+                palindrome.insert(int(len(palindrome)/2), character)
```

Comment:
Why not only do this if the number is odd, say, character_number % 2 == 1? You wouldn't need to insert, as palindrome would be empty at the start.